packages/api-server/src/base/worker.ts

```diff
@@ -0,0 +1,96 @@
+import * as Sentry from "@sentry/node";
+import { envConfig } from "./env-config";
+import { logger } from "./logger";
+const newrelic = require("newrelic");
+
+const POLL_TIME_INTERVAL = 5000; // 5s
+const LIVENESS_CHECK_INTERVAL = 5000; // 5s
+
+// TODO: use the following class to rewrite BlockEmitter
+export class BaseWorker {
+  protected isRunning: boolean;
+  protected pollTimeInterval: number;
+  protected livenessCheckInterval: number;
+
+  constructor({
+    pollTimeInterval = POLL_TIME_INTERVAL,
+    livenessCheckInterval = LIVENESS_CHECK_INTERVAL,
+  } = {}) {
+    this.isRunning = false;
+    this.pollTimeInterval = pollTimeInterval;
+    this.livenessCheckInterval = livenessCheckInterval;
+  }
+
+  // Main worker
+  async startForever() {
+    await this.start();
+    setInterval(async () => {
+      if (!this.running()) {
+        logger.error(
+          `${this.constructor.name} has stopped, maybe check the log?`
+        );
+        await this.start();
+      }
+    }, this.livenessCheckInterval);
+  }
+
+  async start() {
+    this.isRunning = true;
+    this.scheduleLoop();
+  }
+
+  stop() {
+    this.isRunning = false;
+  }
+
+  running() {
+    return this.isRunning;
+  }
+
+  protected scheduleLoop(timeout = 1) {
```

Comment:
Isn't the default 1 millisecond timeout too short?
Consider changing to `scheduleLoop(ms?: number)`, to be the same with `setTimeout`?

Reply:
will do

Reply:
updated